File: examples/quickstart/src/test/java/org/fluentlenium/examples/hooks/ExampleHook.java
package org.fluentlenium.examples.hooks;

import org.fluentlenium.core.FluentControl;
import org.fluentlenium.core.components.ComponentInstantiator;
import org.fluentlenium.core.hook.BaseHook;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.pagefactory.ElementLocator;

import java.util.function.Supplier;

public class ExampleHook extends BaseHook<ExampleHookOptions> {
    public ExampleHook(final FluentControl control, final ComponentInstantiator instantiator,
            final Supplier<WebElement> elementSupplier, final Supplier<ElementLocator> locatorSupplier,
            final Supplier<String> toStringSupplier, final ExampleHookOptions options) {
        super(control, instantiator, elementSupplier, locatorSupplier, toStringSupplier, options);
    }

    @Override
    public void click() {
        System.out.println(getOptions().getMessage() + ": before click!");
        super.click();
        System.out.println(getOptions().getMessage() + "ExampleHook: after click!");
    }
}

File: fluentlenium-kotest/src/test/kotlin/io/fluentlenium/adapter/kotest/hooks/ExampleHook.kt
package io.fluentlenium.adapter.kotest.hooks

import io.fluentlenium.core.FluentControl
import io.fluentlenium.core.components.ComponentInstantiator
import io.fluentlenium.core.hook.BaseHook
import org.openqa.selenium.WebElement
import org.openqa.selenium.support.pagefactory.ElementLocator
import java.util.function.Supplier

class ExampleHook(
    control: FluentControl,
    instantiator: ComponentInstantiator,
    elementSupplier: Supplier<WebElement>,
    locatorSupplier: Supplier<ElementLocator>,
    toStringSupplier: Supplier<String>,
    options: ExampleHookOptions
) :
    BaseHook<ExampleHookOptions>(control, instantiator, elementSupplier, locatorSupplier, toStringSupplier, options) {
    override fun click() {
        super.click()
        clickCount++
    }

    companion object {
        var clickCount = 0
    }
}
